### DISCUSSION TITLE
Mapeamento N-N no EF com um dos lados sendo classe herdada.

### DISCUSSION THREAD

**Discussion by mathfrance** [balta-io/forum#208] [Tira Dúvidas] [answered]
Estou um pouco confuso em como seria a forma correta de criar esse mapeamento no EF, onde: 

Classe Abstrata Pai:
![EntertenimentManagement - Microsoft Visual Studio_2](https://user-images.githubusercontent.com/23667292/210677997-d1f87b9e-5170-4293-9b22-1388145a6f87.jpg)

As 2 classes que herdam de Item:
![EntertenimentManagement - Microsoft Visual Studio_3](https://user-images.githubusercontent.com/23667292/210678371-318110ce-e440-4520-874b-ea2cb8926f4c.jpg)
![EntertenimentManagement - Microsoft Visual Studio_4](https://user-images.githubusercontent.com/23667292/210678416-32095feb-d4e2-404e-94fd-719adb6f07fa.jpg)

E a classe que relaciona com Item:
![EntertenimentManagement - Microsoft Visual Studio_5](https://user-images.githubusercontent.com/23667292/210678513-0b687b62-842b-4e9a-8c4f-8a5e8e0f0257.jpg)

P.s: a Base possui apenas o Id, conforme mostrado no curso.

Pensei em mapear o relacionamento N-N tanto no GameMap quanto no MovieMap em duas novas tabelas vinculando o Id da PersonalList, porém não sei se seria a forma correta visto que originalmente a List<PersonalList> está no Item e não nos filhos, então talvez teria que criar apenas 1 tabela relacional com os dois, e não sei ainda como fazer o mapeamento dos dois na PersonalList (Aliás precisa ou basta mapear em um dos lados?)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @mathfrance, você pode criar um mapeamento para cada item filho, que resulta em uma tabela para cada item.
Ou ter um mapeamento para o base e outros mapeamentos para cada filho, que resulta em uma tabela pai e duas outras filhas.

O primeiro item que citei, vira um "tabelão"... ai você escolhe qual fica melhor para o seu cenário. Não tem certo ou errado.
